### DISCUSSION TITLE
Farming with cold wallet

### DISCUSSION THREAD

**Discussion by racer8** [Chia-Network/chia-blockchain#6849] [Q&A]
To protect my wallet I have created a cold wallet on an offline server.  I then try to farm with the plots are built with farm keys for the cold wallet, the farmer connects to a node that does not contain the private cold keys (for security).  However the farmer (or node) issues a warning that the hot node does not hold the private keys for the plots and refuses to farm with them.  Is there any way of farming with keys that are not online and therefore insecure?

**Comment by loppefaaret** [comment]
i think you are trying to do this the wrong way around - you create a offline cold wallet, and take the receive address from this cold wallet, and input it as the "Farming Reward Target Address" or manually set it through the `config.yaml` at the 2 entries of `xch_target_address: xch1...` and restart chia
the farmer will _always_ need the key that the plots was created for!

**Reply by racer8** [reply]
Thanks - farming node entries for xch_target_address are already set to the cold target address.  The gui still declares the plots as "not found".    The log has "WARNING  Plot plot_name.plot has a farmer public key that is not in the farmer's pk list."

Does the farmer need the public or private key of the target address?

**Reply by loppefaaret** [reply]
the plots hold the public keys, that are linked to the private key - try rebooting, and if it still says it, try running a plots check, and see some of the last lines that it outputs, it might be complaining about something specific about missing entries in config

**Reply by racer8** [reply]
Plots check runs fine with no warnings or errors.
I ask again if the farmer needs the target address private keys because the only fix that works is to add the private key to the farming node wallet which takes me back to my original problem - a hot target wallet.

**Reply by loppefaaret** [reply]
sorry, no. the farmer does not require the "Farming Reward Target Address" private key - it can direct winnings to _any_ receive address here
you should only see "WARNING Plot plot_name.plot has a farmer public key that is not in the farmer's pk list." if the farmer machine does not have a private key for a plot under the harvester running on the local machine

**Reply by racer8** [reply]
Not sure what your last statements means.  My harvester and farmer are running on the same machine - so that machine needs the private key of the target address?  That would explain why the plots work when I add the private key to the famer/harvester machine.

**Reply by loppefaaret** [reply]
it was more a digging statement, that if you were perhaps using a remote harvester, this would be expected behavior - but you are not using a remote harvester - so we need to figure out why it is saying these WARNINGs... 
no, in the case that you have set a cold wallets receive address as the "Farmer Reward Target Address" - the farmer machine do not require the private key for the cold wallet!
the farmer only needs the key to the plot it is managing! and that is exactly what it is complaining about, it is saying that there is plot files that it does not have the private key for! (there are some bugs where this happen, despite your actually having the key imported)
so just to iterate, you need the private key any plot file was created for, imported on the machine running the farmer that manage them
the farmer does not need a private key imported for _any_ receive address set as the "Farmer Reward Target Address" (only in the case that this is a receive address that belongs to the same private key as the plot files was created for - hence not a cold wallet farming setup)

**Reply by racer8** [reply]
So I need the private key on the farmer machine but the farmer does not use it?  Where is the "Farmer Reward Target Address" you mention?    How do I set up a cold wallet farming or is it not possible?

**Reply by loppefaaret** [reply]
https://github.com/Chia-Network/chia-blockchain/wiki/Chia-Keys-Management

**Reply by racer8** [reply]
OK - thanks.
